### ISSUE TITLE
[MNT]: AppVeyor not run on merging to main

### ISSUE THREAD

**Issue by oscargus** [matplotlib/matplotlib#26067]
### Summary

It seems like AppVeyor are only run on PRs, but not on merging to main. Hence, the code coverage is higher in the PRs than on main. This holds especially for Win32-specific parts of the code as well as Gtk4 (which looks more or less untested on main).

### Proposed fix

Enable AppVeyor on merges to main.

**Comment by rcomer** [comment]
Looking back through the last few commits, I think it is running but not on changes that fall under `doc`.  Likely because we have this:

https://github.com/matplotlib/matplotlib/blob/19d93b787649652059bd4163c2b23c10e5088f4f/.appveyor.yml#L12-L16

We have similar for Github Actions and Azure, but in those cases the filter is specifically for PRs:

https://github.com/matplotlib/matplotlib/blob/19d93b787649652059bd4163c2b23c10e5088f4f/.github/workflows/tests.yml#L12-L20

https://github.com/matplotlib/matplotlib/blob/19d93b787649652059bd4163c2b23c10e5088f4f/azure-pipelines.yml#L10-L17

Maybe we need to also make the Appveyor skips specific to PRs.

**Comment by oscargus** [comment]
Good spotting! One could think that we shouldn't run the other tests generating code coverage on doc changes? Although if that ends up with 0% coverage that is of course a worse scenario...

**Comment by rcomer** [comment]
CodeCov has ways to account for skipped tests, but I think I only half understood it - #25391

**Comment by oscargus** [comment]
Hmm , it seems like AppVeyor is the only one correctly skipping tests on doc-only changes?

**Comment by oscargus** [comment]
Nah, both GitHub and Azure skips correctly for PRs, but not for main.
